canonical/snapcraft.io
# update(dep): store-api to 4.0.0 (breaking) including refactor

## Done

- Update canonicalwebteam.store-api to 4.0.0

### Breaking changes
#### `icon_url` is deprecated
- Move `store/logic.py` method `get_icon` to `helpers.py`
- Update `get_icon` to return the icon `url` or an empty string
- Update snaps results to loop through and call `get_icon` for all snaps

### `publisher` ⇾ `developer_name`
- Update all uses of the `publisher` key to `developer_name` key

## How to QA
Pass criteria: Snaps have an icon and (mostly) a publisher name displayed
- Go to the demo
- Visit / make sure the carousel snaps look correct
- Visit /store make sure the snaps on this page look correct
  - Check the featured category page
  - Check another category page, including clicking the "next" button to see the next page
- Search for something e.g. "Music" - make sure everything looks ok
- Visit a distro page /install/spotify/fedora ensure everything looks ok when you scroll down
- Visit a snap ensure everything looks ok
- Go to the edit listings page for one of your snaps, click the preview link, make sure everything looks ok

## Issue / Card
Fixes https://warthogs.atlassian.net/browse/WD-928
